### PR TITLE
Initial PR for package manager

### PR DESCRIPTION
**Pull request by AlexanderSher**
- Some UI
- Folder structure and placeholders

### CODE REVIEW

File: src/Package/Impl/Repl/VsRInteractiveWorkflowProvider.cs [hunk 1]
```diff
@@ -23,11 +25,13 @@ internal class VsRInteractiveWorkflowProvider : IRInteractiveWorkflowProvider {
         [ImportingConstructor]
         public VsRInteractiveWorkflowProvider(IRSessionProvider sessionProvider
             , IRHistoryProvider historyProvider
+            , IRPackageManagerProvider pacakagesProvider
             , IActiveWpfTextViewTracker activeTextViewTracker
             , IDebuggerModeTracker debuggerModeTracker) {
 
             _sessionProvider = sessionProvider;
             _historyProvider = historyProvider;
+            _pacakagesProvider = pacakagesProvider;
```
Comment (huguesv) [inline src/Package/Impl/Repl/VsRInteractiveWorkflowProvider.cs]: typo

File: src/Package/Impl/Resources.resx [hunk 2]
```diff
@@ -522,6 +522,9 @@ Supported formats are: jpg/jpeg, png, tif/tiff, bmp.</value>
   <data name="HistoryWindowCaption" xml:space="preserve">
     <value>R History</value>
   </data>
+  <data name="PackageManagerWindowCaption" xml:space="preserve">
+    <value>R History</value>
```
Comment (huguesv) [inline src/Package/Impl/Resources.resx]: R Packages?

File: src/R/Components/Impl/PackageManager/Implementation/View/InfiniteScrollList.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
+﻿using System;
```
Comment (huguesv) [inline src/R/Components/Impl/PackageManager/Implementation/View/InfiniteScrollList.xaml.cs]: license header missing

File: src/R/Components/Impl/PackageManager/Implementation/View/DesignTime/DesignTimeRPackageInfoListDataSource.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections.Generic;
```
Comment (huguesv) [inline src/R/Components/Impl/PackageManager/Implementation/View/DesignTime/DesignTimeRPackageInfoListDataSource.cs]: License header missing

File: src/R/Components/Impl/PackageManager/Implementation/View/DesignTime/DesignTimeRPackageViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
+﻿using Microsoft.R.Components.PackageManager.Model;
```
Comment (huguesv) [inline src/R/Components/Impl/PackageManager/Implementation/View/DesignTime/DesignTimeRPackageViewModel.cs]: License header missing

File: src/R/Components/Impl/PackageManager/Implementation/View/LoadingStatusBar.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
```
Comment (huguesv) [inline src/R/Components/Impl/PackageManager/Implementation/View/LoadingStatusBar.xaml.cs]: Header again

File: src/R/Components/Impl/PackageManager/Implementation/View/InfiniteScrollList.xaml.cs (hunk 3 above)
Comment (huguesv) [inline src/R/Components/Impl/PackageManager/Implementation/View/InfiniteScrollList.xaml.cs]: i'll stop repeating that comment, please just check all files :)

File: src/R/Components/Impl/PackageManager/Implementation/View/Spinner.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+namespace Microsoft.R.Components.PackageManager.Implementation.View {
+    /// <summary>
+    /// Interaction logic for Spinner.xaml
+    /// </summary>
+    public partial class Spinner : UserControl {
+        public Spinner() {
+            InitializeComponent();
+        }
+
+        public double ScaleX => ActualWidth/120;
```
Comment (huguesv) [inline src/R/Components/Impl/PackageManager/Implementation/View/Spinner.xaml.cs]: nitpick: spaces around operator

File: src/R/Components/Test/Fakes/InteractiveWindow/TestRInteractiveWorkflowProvider.cs [hunk 8]
```diff
@@ -16,6 +17,7 @@ namespace Microsoft.R.Components.Test.Fakes.InteractiveWindow {
     public class TestRInteractiveWorkflowProvider : IRInteractiveWorkflowProvider {
         private readonly IRSessionProvider _sessionProvider;
         private readonly IRHistoryProvider _historyProvider;
+        private readonly IRPackageManagerProvider _pacakagesProvider;
```
Comment (huguesv) [inline src/R/Components/Test/Fakes/InteractiveWindow/TestRInteractiveWorkflowProvider.cs]: typo

File: src/R/Wpf/Impl/Microsoft.R.Wpf.csproj [hunk 9]
```diff
@@ -0,0 +1,80 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (huguesv) [inline src/R/Wpf/Impl/Microsoft.R.Wpf.csproj]: We already have a Microsoft.Common.Wpf project, is this worth creating another project for?

File: src/R/Components/Impl/PackageManager/Model/RPackageVersion.cs [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+namespace Microsoft.R.Components.PackageManager.Model {
+    public class RPackageVersion : IComparable<RPackageVersion> {
+        private readonly string _version;
+
+        public RPackageVersion(string version) {
+            _version = version;
+        }
+
+        public int CompareTo(RPackageVersion other) {
+            return 0;
```
Comment (huguesv) [inline src/R/Components/Impl/PackageManager/Model/RPackageVersion.cs]: I think it's worth writing a TODO comment for this missing implementation.

**Comment by huguesv** [comment]
I'm okay with it.

Note that even there was bad stuff in the xaml files, I probably couldn't tell just looking at the markup (much easier to see it in action).  So I glanced over those.  Most of the UI code is coming from NuGet anyway.

Some parts are clearly not finished, so I didn't bother with notes.

The WPF enum value conversion stuff may be possible to unify with our existing converter?  If so, it can be done later.

File: src/Package/Impl/PackageManager/PackageManagerWindowPane.cs [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+            if (disposing && Component != null) {
+                Component.Dispose();
+                Component = null;
+                _commandTarget = null;
+            }
+            base.Dispose(disposing);
+        }
+
+        public int QueryStatus(ref Guid pguidCmdGroup, uint cCmds, OLECMD[] prgCmds, IntPtr pCmdText) {
+            return _commandTarget.QueryStatus(ref pguidCmdGroup, cCmds, prgCmds, pCmdText);
```
Comment (MikhailArkhipov) [inline src/Package/Impl/PackageManager/PackageManagerWindowPane.cs]: Null check - I've seen VS calling command targets which were disposed... Native/Managed interop issue I think

File: src/Package/Impl/PackageManager/PackageManagerWindowPane.cs [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+        private IOleCommandTarget _commandTarget;
+
+        public PackageManagerWindowPane(IRPackageManager packageManager) {
+            _packageManager = packageManager;
+            Caption = Resources.PackageManagerWindowCaption;
+        }
+
+        protected override void OnCreate() {
+            _commandTarget = new CommandTargetToOleShim(null, Component.Controller);
```
Comment (MikhailArkhipov) [inline src/Package/Impl/PackageManager/PackageManagerWindowPane.cs]: Problem with local controller is that commands are not visible to the top level menu and can only appear as tool window toolbar buttons. For Plot commands I had to redo local commands to PackageCommand (see plot window pane) so they can be visible in the menu

Comment (AlexanderSher) [inline src/Package/Impl/PackageManager/PackageManagerWindowPane.cs]: Yes, but we don't have a toolbar in this window.

File: src/R/Components/Impl/PackageManager/Implementation/View/Spinner.xaml.cs (hunk 7 above)
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/View/Spinner.xaml.cs]: Done

File: src/R/Components/Impl/PackageManager/Model/RPackageVersion.cs (hunk 10 above)
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Model/RPackageVersion.cs]: Ok

File: src/R/Wpf/Impl/Microsoft.R.Wpf.csproj (hunk 9 above)
Comment (AlexanderSher) [inline src/R/Wpf/Impl/Microsoft.R.Wpf.csproj]: I want to split R-Specific styles, brushes, converters etc. from generic things like lambda-converter

File: src/Package/Impl/PackageManager/PackageManagerWindowPane.cs (hunk 11 above)
Comment (AlexanderSher) [inline src/Package/Impl/PackageManager/PackageManagerWindowPane.cs]: Done

**Pull request merged**